File: lab2.6.py
array = []
number_of_elements = int(input())
element = input().split()
for i in element:
    array.append(int(i))
def Sort(list):
    if  len(list) <= 1:
        return list
    left = Sort(list[:len(list) // 2])
    right = Sort(list[len(list) // 2:])
    res_index = 0
    l_index = 0
    r_index = 0
    result_array = [''] * (len(left) + len(right))
    while r_index < len(right) and l_index < len(left):
        if left[l_index] <= right[r_index]:
            result_array[res_index] = left[l_index]
            l_index += 1
        else:
            result_array[res_index] = right[r_index]
            r_index += 1
        res_index += 1
    while l_index < len(left):
        result_array[res_index] = left[l_index]
        res_index += 1
        l_index += 1
    while r_index < len(right):
        result_array[res_index] = right[r_index]
        res_index += 1
        r_index += 1
    for i in range(len(list)):
        list[i] = result_array[i]
    return list
print(*Sort(array))

File: lab2.6.c
#include <stdio.h>

void merge(int mass_first[], int left, int mid, int right)
{
    int it1 = 0;
    int it2 = 0;
    int tmp1[right - left];
    while ((mid + it2 < right) && (left + it1 < mid))
    {
        if (mass_first[left + it1] < mass_first[mid + it2])
        {
            tmp1[it1 + it2] = mass_first[left + it1];
            it1++;
        }
        else
        {
            tmp1[it1 + it2] = mass_first[mid + it2];
            it2++;
        }
    }
    while (mid + it2 < right)
    {
        tmp1[it1 + it2] = mass_first[mid + it2];
        it2++;
    }
    while (left + it1 < mid)
    {
        tmp1[it1 + it2] = mass_first[left + it1];
        it1++;
    }

    for(int i = 0; i < it1 + it2; i++)
    {
        mass_first[left + i] = tmp1[i];
    }
}

void mergeRecursive(int mass_first[], int left, int right)
{
    
    if(left + 1 >= right) return;

    int mid = (left + right) / 2;

    mergeRecursive(mass_first, left, mid);
    mergeRecursive(mass_first, mid, right);
    merge(mass_first, left, mid, right);
}

int main()
{
    int number_of_elements, element;
    scanf("%d", &number_of_elements);
    int array[number_of_elements];
    for (int i = 0; i <= number_of_elements - 1; i ++)
    {
        scanf("%d", &element);
        array[i] = element;
    }
    mergeRecursive(array, 0, number_of_elements);
    for (int i = 0; i < number_of_elements; i ++)
                {
                    if (i == number_of_elements - 1)
                    {
                        printf("%d\n", array[i]);
                    }
                    else
                    {
                        printf("%d ", array[i]);
                    }
                }
    return 0;
}

